src/main/java/org/jfrog/hudson/pipeline/common/executors/BuildAppendExecutor.java

```diff
@@ -0,0 +1,134 @@
+
+        // Prepare Artifactory server
+        org.jfrog.hudson.ArtifactoryServer server = Utils.prepareArtifactoryServer(null, pipelineServer);
+        CredentialsConfig credentialsConfig = server.getResolverCredentialsConfig();
+        Credentials credentials = credentialsConfig.provideCredentials(build.getParent());
+
+        // Calculate build timestamp
+        long timestamp = getBuildTimestamp(server, credentials);
+
+        // Get checksum headers from the build info artifact
+        addChecksumHeaders(server, credentials, moduleBuilder, timestamp);
+
+        // Add module to build info
+        buildInfo.getModules().add(moduleBuilder.build());
+    }
+
+    /**
+     * Get build timestamp of the build to append. The build timestamp have to be converted to seconds from epoch.
+     * For example, start time of: 2020-11-27T14:33:38.538+0200 should be converted to 1606480418538.
+     *
+     * @param server      - The Artifactory server
+     * @param credentials - The credentials of the Artifactory serer
+     * @return the build timestamp of the build to append.
+     * @throws IOException    in case of any communication error with Artifactory, wrong credentials or if the build if missing.
+     * @throws ParseException in case of the returned timestamp contains unexpected format.
+     */
+    private long getBuildTimestamp(org.jfrog.hudson.ArtifactoryServer server, Credentials credentials) throws IOException, ParseException {
+        try (ArtifactoryBuildInfoClient client = server.createBuildInfoClientBuilder(credentials, createProxyConfiguration(), new JenkinsBuildInfoLog(listener)).build()) {
+            Build build = client.getBuildInfo(buildName, buildNumber);
+            DateFormat format = new SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ss.SSSZ");
+            Date date = format.parse(build.getStarted());
+            return date.getTime();
+        }
+    }
+
+    /**
+     * Download MD5 and SHA1 from the build info artifact. Add them to the current build info object.
+     *
+     * @param server        - The Artifactory server
+     * @param credentials   - Credential of the Artifactory server
+     * @param moduleBuilder - The module builder of the new module which will be added to the current build info object
+     * @param timestamp     - The build timestamp of the build to append
+     * @throws IOException if a header the build info object are missing.
+     */
+    private void addChecksumHeaders(org.jfrog.hudson.ArtifactoryServer server, Credentials credentials, ModuleBuilder moduleBuilder, long timestamp) throws IOException {
+        // To support build name and numbers with ':', encode ':' by '%3A'. Further encoding will take place in 'getArtifactMetadata'.
+        String encodedBuildName = buildName.replaceAll(":", "%3A");
+        String encodedBuildNumber = buildNumber.replaceAll(":", "%3A");
```

Comment:
Let's try to add this encoding the build-info's ```encodeUrl``` method, and check if our code lives peacefully with this change, when the build name / build number / other props include ":".

Reply:
We agreed not to change it. the build name and build number have to be encoded separately - we can't encode colon in the entire URL because it contains colon in the protocol.